backend/projects/views.py

```diff
@@ -105,28 +112,34 @@ class ProjectViewSet(PermissionPolicyMixin, viewsets.ModelViewSet):
     }
 
     def create(self, request, *args, **kwargs):
-        # Obecnie, w przypadku gdy nie ma użytkownika z podanym mailem, tworzony jest
-        # nowy ale później można tu zaimplementować rozsyłanie maili
-        members = request.data["members"]
-        members.append(self.request.user.email)
-
-        for member in members:
-            if not User.objects.filter(email=member).exists():
-                User.objects.create_user(
-                    username=member.split("@")[0],
-                    password=member.split("@")[0],
-                    email=member,
+        member_emails = request.data["members"]
+        member_emails.append(self.request.user.email)
+
+        new_users = []
+
+        for member_email in member_emails:
+            if not User.objects.filter(email=member_email).exists():
+                randUsername = "".join(random.choice(string.ascii_letters) for x in range(150))
+                randPass = "".join(random.choice(string.ascii_letters) for x in range(150))
+                serializer = UserSerializer(
+                    data={"username": randUsername, "email": member_email, "password": randPass, "is_active": False}
                 )
+                serializer.is_valid(raise_exception=True)
+                new_users.append(serializer.save())
 
         response = super().create(request, *args, **kwargs)
 
         ownerMember = ProjectUser.objects.filter(
-            project__name=request.data["name"],
-            person__email=request.user.email,
+            project__id=response.data["id"],
+            person__email=self.request.user.email,
         ).first()
         ownerMember.role = "own"
         ownerMember.save()
 
+        for new_user in new_users:
+            link = generate_join_link(new_user.email)
+            get_join_project_email(new_user.email, response.data["name"], link).send()
+
```

Comment:
Tu za każdym razem jak wysyłasz maila to otwierasz połączenie, nie wiem czy nie lepiej utworzyć maile a potem je naraz wysłać, chyba była taka metoda w EmailBuilder